### PR TITLE
Upgraded to Assimp 4.1

### PR DESCRIPTION
**Pull request by NolaDonato**
1. Fixed issue with light rotation
2. Fixed bug importing embedded textures
3. Initialized native number sizes
4. Modified Jassimp IO to conform to new Java api

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
There are some commits with photos.

**Comment by liaxim** [comment]
Is this ready for review and test?

**Comment by NolaDonato** [comment]
yes, it passed all the asset tests. I need help removing the commits. I really don't care that much since these photos will appear on my Siggraph badge and are public.

**Comment by liaxim** [comment]
Switch to your assimp4.1 branch. Run ``git rebase -i HEAD~15``. Comment out (#) unwanted commits and save. Verify the end result is what you expect. Run ``git push origin HEAD:assimp4.1 --force``.

**Comment by liaxim** [comment]
Needs rebase

**Comment by liaxim** [comment]
Are you building assimp off the official tree? The new binaries look to be statically linked to the standard library. Based on https://github.com/assimp/assimp/tree/master/contrib/android-cmake we have a bit of a problem: libc++_shared is officially not supported. We will be subjected to inexplicable crashes like the one Dimitriy encountered yesterday.

I can look into the assimp build and switch to libc++_shared. If that doesn't work maybe exporting only assimp-symbols from libassimp could.

**Comment by liaxim** [comment]
Should we ignore the problems with gvr-meshanimation and gvr-javascript for now and focus on getting this change in first?

**Comment by NolaDonato** [comment]
I will try and fix this today.

**Comment by NolaDonato** [comment]
I believe I have fixed the animation problem. It is ready for test.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRMesh.java [hunk 1]
```diff
@@ -753,9 +753,16 @@ public static GVRMesh createQuad(GVRContext ctx, String vertexDesc, float width,
      */
     public void setBones(List<GVRBone> bones)
     {
+        StringBuffer sb = new StringBuffer();
         mBones.clear();
         mBones.addAll(bones);
+        sb.append("BONES\n");
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRMesh.java]: The debugging traces probably should be removed.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java [hunk 2]
```diff
@@ -942,98 +910,91 @@ public GVRTexture loadCompressedCubemapTexture(GVRAndroidResource resource)
...
+            InputStream stream = resource.getStream();
+            if (stream == null)
             {
-                lastError = e;
-                return null;
+                throw new IOException("Cannot open " + path);
             }
-            finally
+            int read;
+            byte[] data = new byte[1024];
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java]: File will be likely read faster if the chunk was bigger; like 64K.

**Comment by liaxim** [comment]
gvr-widgetviewer: the motorcycle model doesn't show. This is the only problem I encountered. Also I think gvr-widgetviewer starts up slower than before.

**Comment by NolaDonato** [comment]
I figured out why the motorcycle doesnt show. The new assimp imports it as six meshes, the app is expecting one. I modified gvr-widget viewer in PR #608

**Comment by liaxim** [comment]
Looks good. Will update with the binaries using the shared standard library afterwards. Planning to merge on Monday.

**Comment by liaxim** [comment]
We built assimp with api level 24. On Android 6 devices this leads to a crash; @mwitchwilliams noticed that yesterday and I reproduced today. Will update the binaries next week.

```
05-25 11:30:57.815 15098 15098 E AndroidRuntime: FATAL EXCEPTION: main
05-25 11:30:57.815 15098 15098 E AndroidRuntime: Process: org.gearvrf.x3ddemo, PID: 15098
05-25 11:30:57.815 15098 15098 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "__aeabi_memcpy" referenced by "/data/app/org.gearvrf.x3ddemo-1/lib/arm/libassimp.so"...
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at java.lang.Runtime.loadLibrary(Runtime.java:372)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at java.lang.System.loadLibrary(System.java:1076)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at org.gearvrf.GVRActivity.<clinit>(GVRActivity.java:68)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at java.lang.Class.newInstance(Native Method)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.app.Instrumentation.newActivity(Instrumentation.java:1095)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3143)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3417)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.app.ActivityThread.access$1100(ActivityThread.java:226)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1843)
05-25 11:30:57.815 15098 15098 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
```

**Pull request merged**